cardano-node/test/Test/Cardano/Node/Chairman.hs

```diff
@@ -8,43 +9,80 @@ module Test.Cardano.Node.Chairman
+    [ "genesis"
+    , "--genesis-output-dir", tempDir <> "/genesis"
+    , "--start-time", show @Int64 (floor (DTC.utcTimeToPOSIXSeconds startTime))
+    , "--protocol-parameters-file", base <> "/scripts/protocol-params.json"
+    , "--k", "2160"
+    , "--protocol-magic", "459045235"
+    , "--n-poor-addresses", "128"
+    , "--n-delegate-addresses", "7"
+    , "--total-balance", "8000000000000000"
+    , "--avvm-entry-count", "128"
+    , "--avvm-entry-balance", "10000000000000"
+    , "--delegate-share", "0.9"
+    , "--real-pbft"
+    , "--secret-seed", "2718281828"
+    ]
 
-  base <- H.getProjectBase
+  -- Launch cluster of three nodes
+  forM_ [0..2] $ \i -> do
+    si <- H.noteShow $ show @Int i
+    dbDir <- H.noteShow $ tempDir <> "/db/node-" <> si
+    socketDir <- H.noteShow $ tempDir <> "/socket"
+    nodeStdoutFile <- H.noteTempFile tempDir "cardano-node.stdout.log"
+    nodeStderrFile <- H.noteTempFile tempDir "cardano-node.stderr.log"
 
-  dirContents <- liftIO $ IO.listDirectory base
+    H.createDirectoryIfMissing dbDir
+    H.createDirectoryIfMissing socketDir
 
-  H.annotateShow $ dirContents
+    H.copyFile (baseConfig <> "/topology-node-" <> si <> ".json") (tempDir <> "/topology-node-" <> si <> ".json")
+    H.copyFile (baseConfig <> "/config-" <> si <> ".yaml") (tempDir <> "/config-" <> si <> ".yaml")
 
-  (Just hIn, _mOut, _mErr, hProcess) <- H.createProcess =<< H.procNode
-    [ "run"
-    , "--database-path", dbDir
-    , "--socket-path", socketDir <> "/node-2-socket"
-    , "--port", "3002"
-    , "--topology", base <> "/configuration/chairman/defaults/simpleview/topology-node-2.json"
-    , "--config", base <> "/configuration/chairman/defaults/simpleview/config-2.yaml"
-    , "--signing-key", base <> "/configuration/chairman/defaults/simpleview/genesis/delegate-keys.002.key"
-    , "--delegation-certificate", base <> "/configuration/chairman/defaults/simpleview/genesis/delegation-cert.002.json"
-    , "--shutdown-ipc", "0"
-    ]
+    hNodeStdout <- H.evalM . liftIO $ IO.openFile nodeStdoutFile IO.WriteMode
+    hNodeStderr <- H.evalM . liftIO $ IO.openFile nodeStderrFile IO.WriteMode
 
-  H.threadDelay 10000000
+    (Just hIn, _mOut, _mErr, hProcess, _) <- H.createProcess =<<
+      ( ( H.procNode
+          [ "run"
+          , "--database-path", dbDir
+          , "--socket-path", socketDir <> "/node-" <> si <> "-socket"
+          , "--port", "300" <> si <> ""
+          , "--topology", tempDir <> "/topology-node-" <> si <> ".json"
+          , "--config", tempDir <> "/config-" <> si <> ".yaml"
+          , "--signing-key", tempDir <> "/genesis/delegate-keys.00" <> si <> ".key"
+          , "--delegation-certificate", tempDir <> "/genesis/delegation-cert.00" <> si <> ".json"
+          , "--shutdown-ipc", "0"
+          ]
+        ) <&>
+        ( \cp -> cp
+          { IO.std_in = IO.CreatePipe
+          , IO.std_out = IO.UseHandle hNodeStdout
+          , IO.std_err = IO.UseHandle hNodeStderr
```

Comment:
This will redirect `stdout` and `stderr` output to files.

Subsequently, will need to do something more sophisticated to handle test assertions based on the output.